helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithm.java

```diff
@@ -152,69 +179,147 @@ private double getAssignmentNormalizedScore(AssignableNode node, AssignableRepli
         .collect(Collectors.toList());
   }
 
-  private List<AssignableReplica> getOrderedAssignableReplica(ClusterModel clusterModel) {
-    Map<String, Set<AssignableReplica>> replicasByResource = clusterModel.getAssignableReplicaMap();
-    List<AssignableReplica> orderedAssignableReplicas =
-        replicasByResource.values().stream().flatMap(replicas -> replicas.stream())
-            .collect(Collectors.toList());
+  private Map<String, MutablePair<Integer, Integer>> computeOverallClusterUtil(
+      List<AssignableNode> nodes) {
+    Map<String, MutablePair<Integer, Integer>> utilizationMap = new HashMap<>();
+    for (AssignableNode node : nodes) {
+      for (String resourceKey : node.getMaxCapacity().keySet()) {
 
-    Map<String, ResourceAssignment> bestPossibleAssignment =
-        clusterModel.getContext().getBestPossibleAssignment();
-    Map<String, ResourceAssignment> baselineAssignment =
-        clusterModel.getContext().getBaselineAssignment();
+        if (utilizationMap.containsKey(resourceKey)) {
+          Integer newUtil = utilizationMap.get(resourceKey).getLeft() + node.getRemainingCapacity()
```

Comment:
getRemainingCapacity() gives you idle capacity, it seems not what you want. Please double-check.

Reply:
Thanks for the review. I updated the PR based on our offline discussion.